frontend/src/components/gallery/gallery.js

```diff
@@ -0,0 +1,31 @@
+import React, {useState, useEffect} from 'react';
+import './gallery.css'
+import pictureService from '../../services/picture.service';
+
+function Gallery({productId}) {
+    const [shown, setShown] = useState([]);
+    const [pictures, setPicture] = useState([]);
+    function onImageClick(event){
+        if (event.target.tagName.toLowerCase() === "img") setShown(event.target.src);
```

Comment:
Why do you check in this function is event for img ? By default it should be,
Another option sending directly src directly for this image like :
`onClick={() => onImageClick(picture.imageUrl)}`